src/cpu.rs

```diff
@@ -988,11 +988,7 @@ impl<M: Bus> CPU<M> {
 
 impl<M: Bus> core::fmt::Debug for CPU<M> {
     fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-        write!(
-            f,
-            "CPU Dump:\n\nAccumulator: {}",
-            self.registers.accumulator
-        )
+        write!(f, "CPU {{ registers: {:?}", self.registers)
```

Comment:
I think printing all registers makes debugging easier.
Not sure if we also want to dump memory. Probably not. It's "static" anyway.

Reply:
Example output:

```
CPU { registers: Registers { accumulator: 0, index_x: 255, index_y: 0, stack_pointer: StackPointer(255), program_counter: 1529, status: Status(PS_OVERFLOW | PS_UNUSED | PS_BRK | PS_DECIMAL_MODE | PS_DISABLE_INTERRUPTS | PS_ZERO | PS_CARRY) }
CPU { registers: Registers { accumulator: 0, index_x: 255, index_y: 0, stack_pointer: StackPointer(254), program_counter: 1530, status: Status(PS_OVERFLOW | PS_UNUSED | PS_BRK | PS_DECIMAL_MODE | PS_DISABLE_INTERRUPTS | PS_ZERO | PS_CARRY) }
CPU { registers: Registers { accumulator: 0, index_x: 255, index_y: 0, stack_pointer: StackPointer(255), program_counter: 1531, status: Status(0x0) }
CPU { registers: Registers { accumulator: 0, index_x: 255, index_y: 0, stack_pointer: StackPointer(255), program_counter: 1533, status: Status(0x0) }
CPU { registers: Registers { accumulator: 0, index_x: 255, index_y: 0, stack_pointer: StackPointer(255), program_counter: 1535, status: Status(0x0) }
```